src/main/java/com/mraof/minestuck/block/redstone/PlatformBlock.java

```diff
@@ -55,24 +61,86 @@ public BlockRenderType getRenderShape(BlockState state)
 			return BlockRenderType.MODEL;
 	}
 	
+	/**
+	 * Using the generator distance block state and its facing,this function determines if a platform generator is still capable of refreshing this block and deletes the block if it is not supported.
+	 * There must be a powered platform generator at the coordinates established by the generator distance and facing, and there must not be any blocks in the tag PLATFORM_ABSORBING or absorbing mode platform receptacles in between the platform block and its generator
+	 */
+	public static void updateSurvival(BlockState state, World world, BlockPos pos)
+	{
+		if(!world.isClientSide() && state.getBlock() instanceof PlatformBlock)
+		{
+			Direction stateFacing = state.getValue(FACING);
+			BlockPos supportingPos = pos.relative(stateFacing.getOpposite(), state.getValue(GENERATOR_DISTANCE));
+			if(world.isAreaLoaded(supportingPos, 0))
+			{
+				BlockState supportingState = world.getBlockState(supportingPos);
+				
+				checkForAbsorbers(state, world, pos, stateFacing);
+				
+				boolean supportingStateIsGenerator = supportingState.getBlock() instanceof PlatformGeneratorBlock;
+				boolean generatorHasSameFacing = false;
+				boolean generatorHasPower = false;
+				if(supportingStateIsGenerator)
+				{
+					generatorHasSameFacing = supportingState.getValue(PlatformGeneratorBlock.FACING) == stateFacing;
+					generatorHasPower = supportingState.getValue(PlatformGeneratorBlock.POWERED);
+				}
+				
+				if(!supportingStateIsGenerator || (!generatorHasPower || !generatorHasSameFacing))
+					world.setBlockAndUpdate(pos, Blocks.AIR.defaultBlockState()); //removes platform block if there is no platform generator, or if a platform generator is there but it is either not powered or not facing the platform block
+				else
+				{
+					boolean generatorSetToInvisible = supportingState.getValue(PlatformGeneratorBlock.INVISIBLE_MODE);
+					boolean thisSetToInvisible = state.getValue(INVISIBLE);
+					boolean differentInvisibilityStatuses = generatorSetToInvisible != thisSetToInvisible;
```

Comment:
Variable introduction is slightly overdone with `differentInvisibilityStatuses`. I'd suggest inlining this variable into the if-statement.